File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/shared/localgateway.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Tag } from "./tag";



// LocalGateway
/** 
 * Describes a local gateway.
**/
export class LocalGateway extends SpeakeasyBase {
  @SpeakeasyMetadata()
  localGatewayId?: string;

  @SpeakeasyMetadata()
  outpostArn?: string;

  @SpeakeasyMetadata()
  ownerId?: string;

  @SpeakeasyMetadata()
  state?: string;

  @SpeakeasyMetadata({ elemType: Tag })
  tags?: Tag[];
}

File: SDKs/amazonaws.com/ec2/2016-11-15/python/src/sdk/models/shared/localgateway.py
from dataclasses import dataclass, field
from typing import List,Optional
from . import *


@dataclass
class LocalGateway:
    r"""LocalGateway
    Describes a local gateway.
    """
    
    local_gateway_id: Optional[str] = field(default=None)
    outpost_arn: Optional[str] = field(default=None)
    owner_id: Optional[str] = field(default=None)
    state: Optional[str] = field(default=None)
    tags: Optional[List[Tag]] = field(default=None)
    
